src/Microsoft.Azure.Functions.Extensions.Mcp/Abstractions/IMcpResource.cs

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using ModelContextProtocol.Protocol;
+using ModelContextProtocol.Server;
+
+namespace Microsoft.Azure.Functions.Extensions.Mcp;
+
+/// <summary>
+/// Represents a Model Context Protocol (MCP) resource.
+/// </summary>
+internal interface IMcpResource
```

Comment:
A lot of the properties on this interface are mutable. Do we want that? I prefer immutability unless it absolutely needs to be mutable.

Reply:
Yup fair, I think at the moment I don't think they are mutable - perhaps only metadata depending on how we are going to handle that in later PRs. Happy to make them immutability to start and we can modify as needed.